pkg/parser/parser_test.go

```diff
@@ -401,6 +422,103 @@ end
 	}
 }
 
+func TestUnusedErr(t *testing.T) {
+	inputs := map[string]string{
+		`
+x := 1
+`: "line 2 column 1: 'x' declared but not used",
+		`
+x := 1
+if true
+	x := 1
+end
+print x
+`: "line 4 column 2: 'x' declared but not used",
+		`
+x := 1
+if true
+	x := 1
+	print x
+end
+`: "line 2 column 1: 'x' declared but not used",
+		`
+x := 1
+if true
+	print "foo"
+else
+	x := 1
+	print x
+end
+`: "line 2 column 1: 'x' declared but not used",
+		`
+x := 1
+if true
+	print "foo"
+else
+	x := 1
+end
+print x
+`: "line 6 column 2: 'x' declared but not used",
+		`
+x := 1
+if true
+	print "foo"
+else if true
+	x := 1
+end
+print x
+`: "line 6 column 2: 'x' declared but not used",
+		`
+x :=1
+for i := range 10
+	x :=2
```

Comment:
nit: missing space before `2` - and the next few times.

Reply:
thanks done. i need an evy formatter.